packages/realm/src/scripts/build/xcode.ts

```diff
@@ -0,0 +1,141 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2024 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import assert from "node:assert";
+import { execSync, spawnSync } from "node:child_process";
+import path from "node:path";
+import { PACKAGE_PATH } from "./common";
+
+const { env } = process;
+
+export const SUPPORTED_PLATFORMS = [
+  "iphoneos",
+  "iphonesimulator",
+  "macosx",
+  // "tvos",
+  // "tvsimulator",
+  // "watchos",
+  // "watchsimulator",
+  // "xros",
+  // "xrsimulator",
+  // "maccatalyst",
+] as const;
```

Comment:
It should be fairly simple for us to increase support for these platforms too.
It seems to me the non-trivial part of this is to ensure we're testing on those platforms too.

Reply:
tvOS is likely the most interesting one but I have no clue how difficult it is to run tests on tvsimulator

Reply:
I'd say `xros` would also be an interesting platform to support.